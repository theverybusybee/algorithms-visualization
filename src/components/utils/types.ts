import { ElementStates } from "../../types/element-states";

export type TStringArray = string[];

export type TSortingStringArray = {
  value: string;
  type: ElementStates;
};

export type TSortingNumberArray = {
  value: number;
  type: ElementStates;
};

export type TSortingArray = {
  value: any;
  type: ElementStates;
};
