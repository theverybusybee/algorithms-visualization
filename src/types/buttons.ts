export enum SortArrayButtons {
  BubbleSort = "bubbleSort",
  SelectionSort = "selectionSort",
  Ascending = "ascending",
  Descending = "descending",
  NewArray = "New Array",
}

export enum StackAndQueueButtons {
  Add = "add",
  Remove = "remove",
  Clear = "clear",
}

export enum LinkedListButtons {
  AddToHead = "addToHead",
  AddToEnd = "addToEnd",
  RemoveFromEnd = "removeFromEnd",
  RemoveFromHead = "removeFromHead",
  InsertByIndex = "insertByIndex",
  ExtractByIndex = "extractByIndex",
}
