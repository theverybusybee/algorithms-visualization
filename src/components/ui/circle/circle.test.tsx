import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Circle component testing events", () => {
  it("Circle is empty", () => {
    const emptyCircle = renderer.create(<Circle />).toJSON();
    expect(emptyCircle).toMatchSnapshot();
  });

  it("Circle contains letter", () => {
    const circleWithLetter = renderer.create(<Circle letter="B" />).toJSON();
    expect(circleWithLetter).toMatchSnapshot();
  });

  it("Circle has head", () => {
    const circleWitHead = renderer.create(<Circle head="top" />).toJSON();
    expect(circleWitHead).toMatchSnapshot();
  });

  it("Circle has another component in its head attribute", () => {
    const circleWithComponentInHead = renderer
      .create(<Circle head={<Circle />} />)
      .toJSON();
    expect(circleWithComponentInHead).toMatchSnapshot();
  });

  it("Circle has tail", () => {
    const circleWitTail = renderer.create(<Circle tail="bottom" />).toJSON();
    expect(circleWitTail).toMatchSnapshot();
  });

  it("Circle has another component in its tail attribute", () => {
    const circleWithComponentInTail = renderer
      .create(<Circle tail={<Circle />} />)
      .toJSON();
    expect(circleWithComponentInTail).toMatchSnapshot();
  });

  it("Circle has visible index", () => {
    const circleWithIndex = renderer.create(<Circle index={0} />).toJSON();
    expect(circleWithIndex).toMatchSnapshot();
  });

  it("Circle is small", () => {
    const smallCircle = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(smallCircle).toMatchSnapshot();
  });

  it("Circle is in default state", () => {
    const defaultCircle = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(defaultCircle).toMatchSnapshot();
  });

  it("Circle is in changing state", () => {
    const changingCircle = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(changingCircle).toMatchSnapshot();
  });

  it("Circle is in modified state", () => {
    const modifiedCircle = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(modifiedCircle).toMatchSnapshot();
  });
});
