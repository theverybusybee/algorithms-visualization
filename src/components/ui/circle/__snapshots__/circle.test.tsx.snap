// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component testing events Circle contains letter 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    >
      B
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component testing events Circle has another component in its head attribute 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  >
    <div
      className="content  "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle-content"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circleIndex"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component testing events Circle has another component in its tail attribute 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  >
    <div
      className="content  "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle-content"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circleIndex"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component testing events Circle has head 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head"
  >
    top
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component testing events Circle has tail 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail"
  >
    bottom
  </div>
</div>
`;

exports[`Circle component testing events Circle has visible index 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component testing events Circle is empty 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component testing events Circle is in changing state 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   changing"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component testing events Circle is in default state 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component testing events Circle is in modified state 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   modified"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component testing events Circle is small 1`] = `
<div
  className="content  "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle  small default"
    data-testid="circle-pointer"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-pointer-content"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
